packages/lesswrong/components/ea-forum/digest/EditDigest.tsx

```diff
@@ -0,0 +1,545 @@
+      newPosts.push({...postData.post, rating: postData.rating})
+      newPostStatuses[postData.post._id] = {
+        _id: postData.digestPost._id,
+        emailDigestStatus: postData.digestPost.emailDigestStatus ?? 'pending',
+        onsiteDigestStatus: postData.digestPost.onsiteDigestStatus ?? 'pending'
+      }
+    })
+    // sort the list by curated, then suggested for curation, then rating, then karma
+    newPosts.sort((a,b) => {
+      if (a.curatedDate && !b.curatedDate) return -1
+      if (!a.curatedDate && b.curatedDate) return 1
+      if (a.suggestForCuratedUserIds && !b.suggestForCuratedUserIds) return -1
+      if (!a.suggestForCuratedUserIds && b.suggestForCuratedUserIds) return 1
+      // TODO: add this back in once we've implemented the rating
+      // if (a.rating !== b.rating) return b.rating - a.rating
+      return b.baseScore - a.baseScore
+    })
+    setPosts(newPosts)
+    setPostStatuses(newPostStatuses)
+  }, [eligiblePosts]) // eslint-disable-line react-hooks/exhaustive-deps
+  
+  // the digest status of each post is saved on a DigestPost record
+  const { create: createDigestPost } = useCreate({
+    collectionName: 'DigestPosts',
+    fragmentName: 'DigestPostsMinimumInfo',
+  })
+  const { mutate: updateDigestPost } = useUpdate({
+    collectionName: 'DigestPosts',
+    fragmentName: 'DigestPostsMinimumInfo',
+  })
+  
+  // track the table filters
+  const [emailDigestFilter, setEmailDigestFilter] = useState<InDigestStatusOption[]>([...DIGEST_STATUS_OPTIONS])
+  const [onsiteDigestFilter, setOnsiteDigestFilter] = useState<InDigestStatusOption[]>([...DIGEST_STATUS_OPTIONS])
+  const [tagFilter, setTagFilter] = useState<string>('')
+  
+  const handleUpdateEmailDigestFilter = (val: InDigestStatusOption) => {
+    let newFilter = [...emailDigestFilter]
+    if (newFilter.includes(val)) {
+      newFilter = newFilter.filter(v => v !== val)
+    } else {
+      newFilter.push(val)
+    }
+    setEmailDigestFilter(newFilter)
+  }
+  
+  const handleUpdateOnsiteDigestFilter = (val: InDigestStatusOption) => {
+    let newFilter = [...onsiteDigestFilter]
+    if (newFilter.includes(val)) {
+      newFilter = newFilter.filter(v => v !== val)
+    } else {
+      newFilter.push(val)
+    }
+    setOnsiteDigestFilter(newFilter)
+  }
+  
+  const resetFilters = () => {
+    setEmailDigestFilter([...DIGEST_STATUS_OPTIONS])
+    setOnsiteDigestFilter([...DIGEST_STATUS_OPTIONS])
+    setTagFilter('')
+  }
+  
+  /**
+   * When you click on a digest status cell, we rotate the status between
+   * "yes", "maybe", "no", and "pending" (in that order).
+   */
+  const handleClickStatusIcon = async (postId: string, statusField: 'emailDigestStatus'|'onsiteDigestStatus') => {
+    // disable updating statuses while we handle this one,
+    // to avoid creating duplicate DigestPost records
+    setStatusIconsDisabled(true)
+    void updatePostDigestStatus(postId, statusField).then(() => setStatusIconsDisabled(false))
```

Comment:
Oh yeah you're right I do have an extra `async`. I thought I couldn't call `setStatusIconsDisabled` twice since react batches state updates, so it wouldn't ever get set to true? But I tried doing it the way you suggested and it seemed to work, so I guess I must be missing something